# Analysis of mobile apps' profitability
For this project, I need to assist a company that makes free apps in the English language for mobile phones. The company needs to develop a new app, and I will help with suggestions based on the analysis of existing data. 
I used the data of apps in App Store and Google Play Store to define what app genre could be the most profitable. 

Firstly, I open both files using reader() function from imported module CSV.

In [1]:
opened_file1 = open('AppleStore.csv')
opened_file2 = open('googleplaystore.csv')
from csv import reader
read_file1 = reader(opened_file1)
read_file2 = reader(opened_file2)
appst_data = list(read_file1)
google_data = list(read_file2)

I want to understand two datasets in hands, i.e. how many apps are there and columns. Dataquest team suggests to use their function as below.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

Furhther, I want to explore the nature of App Store dataset: the header row and how the data is stored. 

In [3]:
appst_sample_data = explore_data(appst_data, 0, 5, rows_and_columns=True)
print(appst_sample_data) 

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16
None


Code abow shows that the dataset with apps from App Store has 7197 apps (7198 minus header row). Each app has 16 parametres, including id, genres, user rating, etc. 
I want to use the same function to explore the dataset with apps from Google Play Store. 

In [4]:
google_sample_data = explore_data(google_data, 0, 5, rows_and_columns = True)
google_sample_data

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


Code above shows that Google Play Store dataset contains data of 10841 apps. Each app has 13 columns with information about it, including name, category, reiews, etc. 

Both datasets are taken from Kaggle. On the forum dedicated to Google Play Store dataset, I found that one of the entries (10472) is incorrect. I want to check how that entry looks to see deviations from other apps. 

In [5]:
google_data[10473] #considering that I still have a header in that list

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

The app above misses information from the column "Category", which makes all information after the app's name incorrect. I want to remove this app, so it does not affect my further analysis. 

In [6]:
del google_data[10473]

I explored the discussion session of Kaggle a bit more. One of the contributors said that the dataset of Google Play Store had some duplicats. I want to explore it using several popular apps as an example. 

In [7]:
for app in google_data:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [8]:
for app in google_data:
    name = app[0]
    if name == 'Facebook':
        print(app)

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']


It seems that Google Play dataset has more than one duplicate app. I want to explore the number of duplicates in the dataset. 

In [9]:
duplicate_apps = []
unique_apps = []
for app in google_data:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print(len(duplicate_apps))
print(duplicate_apps[:6])

1181
['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box']


Code above shows that Google Play Store dataset has 1181 duplicates. I need to use rule of thumb to decide which entries of the same app I should remove and which should keep. 
Detailed exploration of Instagram entries showed that the main difference between entries is the number of reviews. I assume that the most recent version of an app has the biggest amount of reviews. Thus, for each set of duplicates I want to keep an entry with the biggest amount of reviews. 

In [10]:
reviews_max = {} #firstly, I want to create a dictionary with unique apps
for app in google_data[1:]:
    name = app[0]
    n_reviews = float(app[3]) #all information in the list is given in the string format, so I need to transform it
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews #I rewrite entries the way that only entries with the biggest amount of reviews are left
    if name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))

9659


In [11]:
android_clean = [] #further, I want to have a list of unique apps
already_added = []
for app in google_data[1:]:
    name = app[0]
    n_reviews = float(app[3])
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
google_header = google_data[0]
print(len(android_clean))

9659


Thus, the dataset of unqiue apps from Google Play Store has information about 9659 apps (in the code above I removed the header dataset and added it to the string google_header). 
Before moving to further analysis, I want to check if App Store dataset also has duplicates. The forum says nothing but better safe than sorry. 

In [12]:
duplicate_ap_apps = []
unique_ap_apps = []
for app in appst_data:
    name = app[1]
    if name in unique_ap_apps:
        duplicate_ap_apps.append(name)
    else:
        unique_ap_apps.append(name)
        
print(len(duplicate_ap_apps))
print(duplicate_ap_apps[:6])

2
['Mannequin Challenge', 'VR Roller Coaster']


Apple Store Dataset also has duplicate entries for two apps. I want to explore the difference between several entries. 

In [13]:
for app in appst_data:
    if app[1] == 'Mannequin Challenge':
        print(app)

['1173990889', 'Mannequin Challenge', '109705216', 'USD', '0.0', '668', '87', '3.0', '3.0', '1.4', '9+', 'Games', '37', '4', '1', '1']
['1178454060', 'Mannequin Challenge', '59572224', 'USD', '0.0', '105', '58', '4.0', '4.5', '1.0.1', '4+', 'Games', '38', '5', '1', '1']


I will use the same rule of thumb for App Store dataset: the most recent entry is an entry that has the biggest reviews total rating (the column after price, [5]). I will clean the data the same way: firstly make a dictionary, then add unique apps to a list. 

In [14]:
reviews_app_max = {} 
for app in appst_data[1:]:
    name = app[1]
    n_reviews = float(app[5])
    if name in reviews_app_max and reviews_app_max[name] < n_reviews:
        reviews_app_max[name] = n_reviews
    if name not in reviews_app_max:
        reviews_app_max[name] = n_reviews
print(len(reviews_app_max))

7195


In [15]:
appst_clean = [] 
already_appst_added = []
for app in appst_data[1:]:
    name = app[1]
    n_reviews = float(app[5])
    if (reviews_app_max[name] == n_reviews) and (name not in already_appst_added):
        appst_clean.append(app)
        already_appst_added.append(name)
apple_header = appst_data[0]
print(len(appst_clean))

7195


I removed all duplicates from App Store dataset and derived the header to a separate string. Now I have two lists, appst_clean and android_clean, which have  no duplicates. 

The company I make an analysis for is solely specialized in English language games. There are apps in both Google Play Store and App Store that are not in the English language (for example, Chinese). Having these apps in the analysis will affect validity, since trends could be different for Chinese language users compared to English language users. Thus, I want to remove all applications that have other than English-language symbols in the title. For that, I will consider ASCII (American standard code for information interchange), according to which English letters and symbols have numbers from 0 to 127. I will write a function to define apps that are not in the required range and then apply this function. 

In [16]:
def is_English(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127: #ord() refers to the ASCII system
            non_ascii += 1
    
    if non_ascii > 3: #some apps could have symbols, e.g. emojies, but still have the title with English symbols. To prevent removing valid apps, I add a rule of at least 3 non-ascii charaters
        return False
    else:
        return True

In [17]:
#Now I keep lists only with English-language apps for both stores
android_english = []
ios_english = []

for app in android_clean:
    if is_English(app[0]):
        android_english.append(app)
        
for app in appst_clean:
    if is_English(app[1]):
        ios_english.append(app)

In [18]:
len(ios_english) #want to check how many App Store apps are left for further analysis

6181

In [19]:
len(android_english) #the same

9614

Although the number of apps has already decreased, I need to add another filter. The company I am making this project for is also interested only in free apps; thus, to keep the validity of the results, I need to take into further analysis only free apps. 

In [20]:
android_final = []
ios_final = []

for app in android_english:
    if app[7] == '0':
        android_final.append(app)

for app in ios_english:
    if app[4] == '0.0':
        ios_final.append(app)

In [21]:
len(android_final)

8864

In [22]:
len(ios_final)

3220

As given by this guided project's assignment, the company I work for wants to start with making a base app for Android. If the app is profitable, they will further develop it; if within 6 months is is still profitable, they will make this app for iOs as well. In the end, both stores should have this app. Therefore, I need to find a genre that is popular both in App Store and Google Play Store.

I will make a function that creates a frequency table for popular genres. 

In [23]:
def freq_table(dataset, index):
    freq_dictionary = {} #I start with counting the number of apps of each genre
    for row in dataset:
        value = row[index]
        if value in freq_dictionary:
            freq_dictionary[value] += 1
        else:
            freq_dictionary[value] = 1
    
    freq_perc_dictionary = {} #I want to transform it into the frequency table with percentages
    total = 0
    for key in freq_dictionary:
        count = freq_dictionary[key]
        total += count #I add the amount of games from each genre
    
    for key in freq_dictionary:
        percentage = freq_dictionary[key] / total * 100 #I transform the number into percentages
        freq_perc_dictionary[key] = percentage
    
    return freq_perc_dictionary

def display_table(dataset, index): #I want to sort keys in frequency table in order of their values
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Once the function is defined, I want to see what app genres are more popular in App Store. 

In [24]:
ios_populargen = display_table(ios_final, 11)
ios_populargen

Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


The most popular genre among developers of ENG free aps on iOs is Games. Such genres as Entertainment, Photo & Video, and social networking are following Games in the number of apps in the store. Catalogs, medical, and navigation apps are least popular among developers (meaning that there is the smallest amount of apps for these genres). 
It seems that there are more apps in App Store for fun applications. However, it does not mean that more users prefer fun apps.

In [25]:
android_popcat = display_table(android_final, 1)
android_popcat

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In Google Play Store, Family, games, and tools are most popular categories among app developers. Business and lifestyle are also close to the top; it seems that apps for not-fun purposes are more popular in Google Play Store than in App Store.

In [26]:
android_popgen = display_table(android_final, 9)
android_popgen

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The most popular genres among free Google Play Store apps are tools, entertainment, education, business, and productivity. This information supports my hypothesis about practical apps being more popular (at least among developers) in Google Play Store compared to App Store. 
However, I need to see what is actually more popular among users. To do so, I need to focus on the columns indicating number of installs. Google Play Store has such data, while App Store only has the total count of ratings as the closest to installations column. Using this logic, I will calculate number of installs for different stores per genre. 

In [27]:
genres_ios = freq_table(ios_final, -5) #I firstly make the frequency table for genres

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[11]
        if genre_app == genre:
            user_ratings = float(app[5])
            total += user_ratings
            len_genre += 1
    avg_ratings_genre = total / len_genre
    print(genre, ":", avg_ratings_genre)

Navigation : 86090.33333333333
Catalogs : 4004.0
Travel : 28243.8
Finance : 31467.944444444445
Photo & Video : 28441.54375
Health & Fitness : 23298.015384615384
Shopping : 26919.690476190477
News : 21248.023255813954
Education : 7003.983050847458
Weather : 52279.892857142855
Music : 57326.530303030304
Medical : 612.0
Reference : 74942.11111111111
Food & Drink : 33333.92307692308
Business : 7491.117647058823
Games : 22812.92467948718
Lifestyle : 16485.764705882353
Productivity : 21028.410714285714
Sports : 23008.898550724636
Entertainment : 14029.830708661417
Book : 39758.5
Utilities : 18684.456790123455
Social Networking : 71548.34905660378


In App Store, apps with genre Navigation have the biggest amount of installations. Next place is Reference, followed by Social networks, Music, and Weather. I want to explore what are the most popular apps in those genres. 

In [28]:
for app in ios_final:
    if app[11] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Navigation is dominated by Waze and Google Maps; even the third app is not installed as often. It has two consequences: it might be hard to compete with such giants and the overall category is not as popular as it seems based on the amount of ratings alone. 

In [29]:
for app in ios_final:
    if app[11] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Reference is dominated by religious books, dictionaries and guides to games (Minecraft and Pokemon). It is an interesting idea for an app - to make an entertaining app with the reference category. 

In [31]:
for app in ios_final:
    if app[11] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

Social apps are dominated by Facebook, Pinterest, Skype, Messenger, and Tumblr. There are also a lot of apps in this category, which might make it harder to compete. 

Now I have an overall idea of App Store data, so I want to analyze Google Play Store data. Firstly, I will do some exploration of the installs column.

In [32]:
display_table(android_final, 5) 

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


It seems that the data provided in a way which requires me to do some additional calculations in the code (because I cannot count these strings, and additional symbols will not allow me to transform them into floats the way they are). I will further use the same method as before to define the most popular apps in Google Play Store based on the amount of installs. 

In [33]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category == category_app:
            number_installs = app[5]
            number_installs = number_installs.replace('+', '') #first part of transforming string
            number_installs = number_installs.replace(',', '') #second part of transforming string
            total += float(number_installs)
            len_category += 1
    avg_installs_number = total / len_category
    print(category, ":", avg_installs_number)

COMMUNICATION : 38456119.167247385
SPORTS : 3638640.1428571427
BEAUTY : 513151.88679245283
HOUSE_AND_HOME : 1331540.5616438356
GAME : 15588015.603248259
BUSINESS : 1712290.1474201474
FOOD_AND_DRINK : 1924897.7363636363
SOCIAL : 23253652.127118643
LIFESTYLE : 1437816.2687861272
MAPS_AND_NAVIGATION : 4056941.7741935486
NEWS_AND_MAGAZINES : 9549178.467741935
EDUCATION : 1833495.145631068
BOOKS_AND_REFERENCE : 8767811.894736841
TOOLS : 10801391.298666667
SHOPPING : 7036877.311557789
LIBRARIES_AND_DEMO : 638503.734939759
DATING : 854028.8303030303
PERSONALIZATION : 5201482.6122448975
AUTO_AND_VEHICLES : 647317.8170731707
PRODUCTIVITY : 16787331.344927534
VIDEO_PLAYERS : 24727872.452830188
PARENTING : 542603.6206896552
FINANCE : 1387692.475609756
TRAVEL_AND_LOCAL : 13984077.710144928
MEDICAL : 120550.61980830671
WEATHER : 5074486.197183099
ART_AND_DESIGN : 1986335.0877192982
PHOTOGRAPHY : 17840110.40229885
ENTERTAINMENT : 11640705.88235294
COMICS : 817657.2727272727
EVENTS : 253542.222222222

I can see that the most popular apps in Google Play Store are Communication, Video Players, Photography, Productivity, Games, Travel and local, Tools, Entertainment, and Reference. I do think that most of the categories are in fact not that popular and are dominated by huge giants. I will test this assumption below. 

In [34]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

I want to see what remains in the Communication genre once the biggest apps are removed. 

In [35]:
under_100m = []
for app in android_final:
    number_installs = app[5]
    number_installs = number_installs.replace(',', '')
    number_installs = number_installs.replace('+', '')
    if float(number_installs) < 100000000 and (app[1] == 'COMMUNICATION'):
        under_100m.append(float(number_installs))

sum(under_100m) / len(under_100m)

3603485.3884615386

The number of installs went down from 38mil to 3.6mil for the category, meaning that the big apps are distorting the resul, changing the skewness of the apps distribution. I want to see if similiar is truth for Video Players and Refreneces. I will transform my code into function.

In [36]:
def most_popular(dataset, genre):
    for app in dataset:
        if app[1] == genre and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
            return (print(app[0], ':', app[5]))

In [37]:
vp_pop = most_popular(android_final, 'VIDEO_PLAYERS')

YouTube : 1,000,000,000+


In [38]:
under_100mvp = []
for app in android_final:
    number_inst = app[5]
    number_inst = number_inst.replace('+', '')
    number_inst = number_inst.replace(',', '')
    if (float(number_inst) < 100000000) and (app[1] == 'VIDEO_PLAYERS'):
        under_100mvp.append(float(number_inst))

sum(under_100mvp) / len(under_100mvp)

5544878.133333334

The situation is almost similiar: after removing the most popular apps, the amount of installs dropped from 24m to 5.5m, almost five times. It means that if the company decides to make a video editing app, they either need to compete with huge giants or not to expect too many installs.

In [39]:
ref_pop = most_popular(android_final, 'BOOKS_AND_REFERENCE')

Google Play Books : 1,000,000,000+


In [40]:
under_100ref = []
for app in android_final:
    number_inst = app[5]
    number_inst = number_inst.replace('+', '')
    number_inst = number_inst.replace(',', '')
    if float(number_inst) < 100000000 and (app[1] == "BOOKS_AND_REFERENCE"):
        under_100ref.append(float(number_inst))

sum(under_100ref) / len(under_100ref)

1437212.2162162163

In [41]:
travel_pop = most_popular(android_final, 'TRAVEL_AND_LOCAL')

Booking.com Travel Deals : 100,000,000+


In [42]:
under_100trav = []
for app in android_final:
    number_inst = app[5]
    number_inst = number_inst.replace('+', '')
    number_inst = number_inst.replace(',', '')
    if float(number_inst) < 100000000 and (app[1] == "TRAVEL_AND_LOCAL"):
        under_100trav.append(float(number_inst))

sum(under_100trav) / len(under_100trav)

2944079.6336633665

One idea could be to make a reference application. The most popular are dedicated either to religious texts or games. If choosing this path, I would suggest makin a reference app for a game. 

Another way is to go for a travel and local guide, which can be stored in App Store as a reference. It seems that the category of travel is relatively popular in Google Play Store, and there is only one indeed big application (Booking). Thus, it is a possibility to create something dedicated and useful, while keeping it playful enough to satisfy App Store's ausience interests. 